### PR TITLE
fix : implemented setting fragment inside setting activity

### LINKED ISSUE #1196: settings cannot be opened from any activity except main activity
**Actual Behaviour**

The settings are only be opened by the Main Activity as it is implemented as a fragment

**Expected Behaviour**

The settings fragment should be inside an activity so that it can be easily launched by any other activity like lux meter activity through the menu and also through the navigation menu from the main activity.

This is required as GPS on/off switch can be created inside settings which should be able to get opened by sensor activities.

**Steps to reproduce it**

Create the setting activity and put the setting fragment inside it.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #1196 
First steps in #1195 

**Changes**: 

- Created new setting activity
- Implemented setting fragment inside setting activity
- Created setting option in menu for lux meter
- Clicking setting in navigation now opens setting activity

Now setting activity can be opened through any activity by simply using intents.

**To do in further PR**

- Create checkbox option in settings fragment to include the location in the csv file for sensors

**Screenshot/s for the changes**: 
setting option in overflow menu
image

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[avi_setting.zip](https://github.com/fossasia/pslab-android/files/2173511/avi_setting.zip)

### CODE REVIEW

**Comment by harsh-2711** [comment]
@Avjeet Please don't make changes to other developers lines in Android manifest. Also, can you please explain why did you make settings fragment because settings activity is enough as we don't have to change layouts inside the settings activity?

**Comment by Avjeet** [comment]
@yatri1609 When the previous PR is merged I will rebase into one commit.

**Comment by Avjeet** [comment]
@harsh-2711 well here in this [link](https://developer.android.com/guide/topics/ui/settings#Fragment) it says to implement the setting fragment inside activity so did that

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -61,25 +61,22 @@
...
-            android:name=".activity.PowerSourceActivity"
-            android:windowSoftInputMode="stateHidden|adjustResize"
+            android:name=".activity.MultimeterActivity"
             android:screenOrientation="portrait" />
-
-        <activity android:name=".activity.WaveGeneratorActivity"
-            android:screenOrientation="userLandscape"/>
-
+        <activity
+            android:name=".activity.PowerSourceActivity"
```
Comment (Vikum94) [inline app/src/main/AndroidManifest.xml]: Why there are changes to these instruments?

File: app/src/main/java/org/fossasia/pslab/activity/MainActivity.java [hunk 2]
```diff
@@ -185,7 +184,7 @@ private Fragment getHomeFragment() throws IOException {
             case 1:
                 return HomeFragment.newInstance(ScienceLabCommon.scienceLab.isConnected(), ScienceLabCommon.scienceLab.isDeviceFound());
             case 2:
-                return SettingsFragment.newInstance();
+                /**/
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/activity/MainActivity.java]: break or return from here. There is no point in checking other cases once it's here isn't it?

File: app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentData.java [hunk 3]
```diff
@@ -24,15 +25,14 @@
 
 import org.fossasia.pslab.R;
 import org.fossasia.pslab.activity.LuxMeterActivity;
-import org.fossasia.pslab.communication.sensors.BH1750;
-
-import java.io.IOException;
-import java.text.DecimalFormat;
-
 import org.fossasia.pslab.communication.ScienceLab;
+import org.fossasia.pslab.communication.sensors.BH1750;
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentData.java]: These changes are not necessary

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Avjeet) [reply app/src/main/AndroidManifest.xml]: This is due to repositioning of code as I added the setting activity tag below main activity

**Comment by CloudyPadmal** [comment]
Yes I guess implementing settings in an activity makes it easier to access it easier. Otherwise have to go through a big mess trying to access from Main activity. Is that right @Avjeet and @harsh-2711 ?

**Comment by Avjeet** [comment]
@CloudyPadmal conflicts resolved

**Comment by harsh-2711** [comment]
I agree with @CloudyPadmal

File: app/src/main/java/org/fossasia/pslab/activity/LuxMeterActivity.java [hunk 4]
```diff
@@ -237,11 +240,12 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
     @Override
     protected void onResume() {
         super.onResume();
-        if (isloggingLocation) {
+        if (checkGpsOnResume) {
             if (gpsLogger.isGPSEnabled()) {
                 saveData = true;
                 Toast.makeText(getApplicationContext(), getString(R.string.data_recording_start),
                         Toast.LENGTH_SHORT).show();
+                checkGpsOnResume=false;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/LuxMeterActivity.java]: Let's do some refactoring here

**Comment by Avjeet** [comment]
@CloudyPadmal changes done
@harsh-2711 @abhinavraj23 @Vikum94 plz review it.

**Comment by Avjeet** [comment]
@CloudyPadmal conflicts resolve.

**Comment by CloudyPadmal** [comment]
@yatri1609 @harsh-2711 @Vikum94 @abhinavraj23 please review this soon as many other PRs depend on this.

**Comment by Avjeet** [comment]
@harsh-2711 @abhinavraj23 @cweitat Plz review this all other PRs are dependent on this.

**Comment by cweitat** [comment]
@Vikum94 are you going to review this?

File: app/src/main/res/xml/settings_preference_fragment.xml [hunk 5]
```diff
@@ -1,18 +1,20 @@
 <?xml version="1.0" encoding="utf-8"?>
 <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android">
-    <PreferenceCategory android:title="App Settings">
+    <PreferenceCategory android:title="General">
```
Comment (harsh-2711) [inline app/src/main/res/xml/settings_preference_fragment.xml]: @Avjeet Don't you think "App Settings" is more suitable as a title?

Comment (Avjeet) [reply app/src/main/res/xml/settings_preference_fragment.xml]: @harsh-2711 almost all the apps have the Settings as there title and moreover if I change the title to app setting I also have to change the name in the navigation panel and in overflow menu in lux meter activity and also it doesn't make much of the difference

File: app/src/main/res/xml/settings_preference_fragment.xml [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
-        android:entryValues="@array/export_data_format_values"
-        android:key="export_data_format_list"
-        android:title="Export Data Format" />
+    <PreferenceCategory android:title="Data Logging">
+        <ListPreference
+            android:defaultValue="0"
+            android:entries="@array/export_data_format_list"
+            android:entryValues="@array/export_data_format_values"
+            android:key="export_data_format_list"
+            android:title="Export Data Format" />
```
Comment (CloudyPadmal) [inline app/src/main/res/xml/settings_preference_fragment.xml]: Can we fetch this from `strings.xml`?

**Comment by Avjeet** [comment]
@CloudyPadmal changes done, conflicts resolved, plz review

**Comment by CloudyPadmal** [comment]
@Avjeet now you can fix the other PRs which were dependent on this

**Pull request merged**